src/ParallelAlgorithms/Actions/MonitorMemory.hpp

```diff
@@ -302,4 +302,47 @@ struct ProcessArray {
         std::make_tuple(time, size_per_node, avg_size));
   }
 };
+
```

Comment:
I don't really love one file with so many different actions in it. My impression is the actions are all meant to be used in different places (maybe one or two in the same place). I think it's clear add either a subdirectory `MemoryMonitor` with several different files, or maybe something like `ProcessSingletonMemory.hpp`?

Reply:
Besides tests, all these actions are going to be using in a `MonitorMemory` event I'll add in a future PR. `ContributeMemoryData` won't be called directly by the event but indirectly by the `ProcessGroup` and `ProcessNodegroup` actions.

That being said, this file is a bit crowded with different things and may be confusing. I can split it up into 4 files in a `MemoryMonitor` subdirectory:
- `ContributeMemoryData.hpp`
- `ProcessGroups.hpp` (has both group and nodegroup since they are so similar)
- `ProcessArray.hpp`
- `ProcessSingleton.hpp`

Reply:
Okay, that sounds good to me if that sounds like the right thing to you. You know all the followup PRs :)